File: app/src/main/java/com/fastaccess/provider/timeline/handler/UnderlineHandler.java
package com.fastaccess.provider.timeline.handler;

import android.text.SpannableStringBuilder;
import android.text.style.UnderlineSpan;

import net.nightwhistler.htmlspanner.TagNodeHandler;

import org.htmlcleaner.TagNode;

public class UnderlineHandler extends TagNodeHandler {

    @Override public void handleTagNode(TagNode tagNode, SpannableStringBuilder spannableStringBuilder, int start, int end) {
        spannableStringBuilder.setSpan(new UnderlineSpan(), start, end, 33);
    }
}

File: markdown/src/main/java/com/fastaccess/markdown/spans/UnderlineHandler.kt
package com.fastaccess.markdown.spans

import android.text.SpannableStringBuilder
import android.text.style.UnderlineSpan
import net.nightwhistler.htmlspanner.SpanStack

import net.nightwhistler.htmlspanner.TagNodeHandler

import org.htmlcleaner.TagNode

class UnderlineHandler : TagNodeHandler() {
    override fun handleTagNode(node: TagNode?, builder: SpannableStringBuilder?, start: Int, end: Int, spanStack: SpanStack?) {
        builder?.setSpan(UnderlineSpan(), start, end, 33)
    }
}